templates/workloads/partials/cf/publish.yml

```diff
@@ -0,0 +1,31 @@
+{{- if .Publish}}
+{{- range $topic := .Publish.Topics}}
+{{$topic.Name}}SNSTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: ${AWS::StackName}-{{$topic.Name}}-SNSTopic
+      
+  {{$topic.Name}}SNSTopicPolicy:
+    Type: AWS::SNS::TopicPolicy
+    DependsOn: {{$topic.Name}}SNSTopic
+    Properties:
+      Topics: 
+        - !Ref ${{.Name}}SNSTopic
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          {{- range $worker := .AllowedWorkers}}
+          - Effect: Allow
+            Principal:
+              Service: sqs.amazonaws.com
+            Action: 
+              - sns:Subscribe
+            Resource: !Sub 'arn:${AWS::Partition}:sqs:${AWS::Region}:${AWS::AccountId}:*'
+            Condition:
+              StringEquals:
+                'sqs:ResourceTag/copilot-application': !Sub '${AppName}'
+                'sqs:ResourceTag/copilot-environment': !Sub '${EnvName}'
+                'sqs:ResourceTag/copilot-service': !Sub {{$worker}}
+          {{- end}}
```

Comment:
Excellent! 😍 Can you work with @bvtujo to ensure that AppSec is aware of these changes 🙏 ? We'd like to start that process as early as possible

Reply:
It looks like that is one of Austin's in progress tasks!